tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Generators/ReadMeGeneratorTool.cs

```diff
@@ -0,0 +1,215 @@
+
+            var chatClient = openAiClient.GetChatClient(model);
+
+            await Generate(
+                chatClient: chatClient,
+                engPath: engPath,
+                templatePath: templatePath,
+                serviceDocumentation: new Uri(serviceDocumentation),
+                packagePath: packagePath,
+                outputPath: outputPath);
+        }
+
+        async Task Generate(ChatClient chatClient, string templatePath, Uri serviceDocumentation, string packagePath, string outputPath, string engPath)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(templatePath);
+            ArgumentNullException.ThrowIfNull(serviceDocumentation);
+            ArgumentException.ThrowIfNullOrEmpty(outputPath);
+            ArgumentException.ThrowIfNullOrEmpty(engPath);
+
+            var readmeText = await File.ReadAllTextAsync(templatePath);
+
+            var messages = new ChatMessage[]{
+                new SystemChatMessage(
+                    string.Join("\n",
+                        "We're going to create some READMEs.",
+                        "The parameters are:",
+                        "* A URL that contains service documentation, to be used when creating key concepts, an introduction blurb and any other places where conceptual docs are needed",
+                        "* A package path that we can use to generate documentation links",
+                        "Here are some more rules to follow:",
+                        "- Do not touch the following sections, or its subsections: Contributing.",
+                        "- Do not generate sample code.",
+                        "- Rules for proper readmes can be found here: https://github.com/Azure/azure-sdk/blob/main/docs/policies/README-TEMPLATE.md"
+                    )
+                ),
+                new SystemChatMessage("The readme template is this: \"" + readmeText + "\" which we fill in with the user parameters, which follow:"),
+                new UserChatMessage($"Service URL: {serviceDocumentation}"),
+                new UserChatMessage($"Package path: {packagePath}")
+            };
+
+            var response = await chatClient.CompleteChatAsync(messages);
+            var generatedReadmeText = Customize(response.Value.Content[0].Text);
+
+            await File.WriteAllTextAsync(outputPath, generatedReadmeText);
+            await Validate(engPath: engPath, readmePath: outputPath);
+
+            output.Output($"Readme written to {outputPath}");
+        }
+
+        private async Task Validate(string engPath, string readmePath)
+        {
+            output.Output($"Running {engPath}/common/scripts/Verify-Links.ps1 \"{readmePath}\"");
+
+            // TODO: I think chradek has a replacement for this.
+            var process = Process.Start(new ProcessStartInfo()
+            {
+                FileName = Environment.OSVersion.Platform == PlatformID.Win32NT ? "powershell.exe" : "pwsh",
```

Comment:
Change this over to [RuntimeInformation.IsOSPlatform](https://learn.microsoft.com/dotnet/api/system.runtime.interopservices.runtimeinformation.isosplatform?view=net-9.0).   It's a lot more reliable and consistent.

Reply:
Done!

I would be interested in what makes the other one unreliable. Maybe tricky because you could get down into comparing windows versions, which I never want to do again.